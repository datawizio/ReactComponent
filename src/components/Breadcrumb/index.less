@import "~antd/es/breadcrumb/style/index";
@import "../../styles/variables";

.dw-breadcrumb.ant-breadcrumb {
  font-size: 20px;
  .ant-breadcrumb-separator {
    display: inline-flex;
    margin: 0 12px;
  }
}
