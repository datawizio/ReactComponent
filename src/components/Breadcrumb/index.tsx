import * as React from "react";

import { Breadcrumb as AntBreadcrumb } from "antd";
import { BreadcrumbProps as AntBreadcrumbProps } from "antd/lib/breadcrumb";
import { LeftOutlined, RightOutlined } from "@ant-design/icons";

import "./index.less";
import { useContext } from "react";
import ConfigContext from "../ConfigProvider/context";

export interface BreadcrumbProps extends AntBreadcrumbProps {}

export type FCBreadcrumbProps = React.FC<BreadcrumbProps>;

export interface FCBreadcrumb extends FCBreadcrumbProps {
  Item: typeof AntBreadcrumb.Item;
  Separator: typeof AntBreadcrumb.Separator;
}

const Breadcrumb: FCBreadcrumb = ({ separator, ...props }) => {
  const { direction } = useContext(ConfigContext);
  if (!separator) separator = direction === "rtl" ? <LeftOutlined /> : <RightOutlined />;
  return (
    <AntBreadcrumb className="dw-breadcrumb" separator={separator} {...props} />
  );
};

Breadcrumb.Item = AntBreadcrumb.Item;
Breadcrumb.Separator = AntBreadcrumb.Separator;

export default Breadcrumb;
