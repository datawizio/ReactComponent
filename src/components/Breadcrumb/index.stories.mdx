import { Meta, Preview, Story } from "@storybook/addon-docs/blocks";
import { PieChartOutlined, DesktopOutlined, ContainerOutlined } from "@ant-design/icons";

import Breadcrumb from "./index";

export const breadcrumbWithIconStyles = {
  display: "inline-flex", 
  alignItems: "baseline", 
  gap: 5
};

<Meta title="Components/Breadcrumb" component={Breadcrumb} />

# Breadcrumb

The `Breadcrumb` component provides a way to display a breadcrumb navigation system.  
It is based on the [**Ant Design Breadcrumb**](https://ant.design/components/breadcrumb/) component.

## Basic

<Preview>
  <Story name="Basic">
    <Breadcrumb>
      <Breadcrumb.Item>Home</Breadcrumb.Item>
      <Breadcrumb.Item>Category</Breadcrumb.Item>
      <Breadcrumb.Item>Item</Breadcrumb.Item>
    </Breadcrumb>
  </Story>
</Preview>

## With Custom Separator

<Preview>
  <Story name="With Custom Separator">
    <Breadcrumb separator="/">
      <Breadcrumb.Item>Home</Breadcrumb.Item>
      <Breadcrumb.Item>Category</Breadcrumb.Item>
      <Breadcrumb.Item>Item</Breadcrumb.Item>
    </Breadcrumb>
  </Story>
</Preview>

## With Icons

<Preview>
  <Story name="With Icons">
    <Breadcrumb>
      <Breadcrumb.Item style={breadcrumbWithIconStyles}>
        <PieChartOutlined />
        <span>Dashboard</span>
      </Breadcrumb.Item>
      
      <Breadcrumb.Item style={breadcrumbWithIconStyles}>
        <DesktopOutlined />
        <span>Workspace</span>
      </Breadcrumb.Item>
      
      <Breadcrumb.Item style={breadcrumbWithIconStyles}>
        <ContainerOutlined />
        <span>Projects</span>
      </Breadcrumb.Item>
    </Breadcrumb>
  </Story>
</Preview>

## Props

The `Breadcrumb` component accepts all the props of the [**Ant Design Breadcrumb**](https://ant.design/components/breadcrumb/#API).
