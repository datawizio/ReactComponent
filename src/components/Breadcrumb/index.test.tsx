import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import Breadcrumb from "./index";

const mockProps = {
  className: "customClassName"
};

const setUp = (props?) => mount(<Breadcrumb {...props} />);

describe("Breadcrumb component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Breadcrumb rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("Breadcrumb with  correct separator", () => {
    const wrapper = setUp({ ...mockProps, separator: ">" });
    expect(wrapper.props().separator).toBe(">");
  });

  it("Breadcrumb get correct className", () => {
    expect(component.find(`.${mockProps.className}`).length).toBeTruthy();
  });
});
