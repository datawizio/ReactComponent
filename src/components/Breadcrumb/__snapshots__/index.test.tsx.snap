// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Breadcrumb rendered correctly 1`] = `
<Breadcrumb
  className="customClassName"
>
  <Breadcrumb
    className="customClassName"
    separator={<ForwardRef(RightOutlined) />}
  >
    <div
      className="customClassName ant-breadcrumb"
    />
  </Breadcrumb>
</Breadcrumb>
`;
